test/arrayops.jl

```diff
@@ -88,13 +88,50 @@ a = reshape(b, (2, 2, 2, 2, 2))
 @test a[2,2,2,2,2] == b[end]
 
 # reshaping linearslow arrays
-a = zeros(1, 5)
+a = collect(reshape(1:5, 1, 5))
 s = sub(a, :, [2,3,5])
-@test length(reshape(s, length(s))) == 3
+r = reshape(s, length(s))
+@test length(r) == 3
+@test r[1] == 2
+@test r[3,1] == 5
+@test r[Base.ReshapedIndex(CartesianIndex((1,2)))] == 3
+@test parent(reshape(r, (1,3))) === r.parent === s
+@test parentindexes(r) == (1:1, 1:3)
+@test reshape(r, (3,)) === r
+r[2] = -1
+@test a[3] == -1
 a = zeros(0, 5)  # an empty linearslow array
 s = sub(a, :, [2,3,5])
 @test length(reshape(s, length(s))) == 0
 
+@test reshape(1:5, (5,)) === 1:5
+@test reshape(1:5, 5) === 1:5
+
+# setindex! on a reshaped range
+a = reshape(1:20, 5, 4)
+for idx in ((3,), (2,2), (Base.ReshapedIndex(1),))
+    try
+        a[idx...] = 7
```

Comment:
if this is supposed to always fail, needs an `error("unexpected")` or something so it doesn't silently pass